vmm/src/lib.rs

```diff
@@ -3388,6 +3462,40 @@ mod tests {
             .is_some());
     }
 
+    #[test]
+    fn test_configure_vmm_from_json() {
```

Comment:
When testing `configure_from_json`, please try to test all branches of the function. Use valid and invalid resources for each resource and `assert` on the expected errors on the invalid types.

Unfortunately we can't actually boot a microVM (see commit `1754f991`) so we might not have any way to test this function unless you decouple the configuration code from the `start_microvm()` so that we can at least test the configuration part.

Reply:
Ok, i decoupled the configuration code and added more unit tests. Now `configure_from_json()` is fully covered, but the code coverage decreased due to the fact that the newly added lines in `start_vmm_thread()` function are hard to test.